src/spartan_config.erl

```diff
@@ -31,17 +31,29 @@ udp_port() ->
 bind_interface() ->
     application:get_env(?APP, bind_interface, undefined).
 
+-spec(bind_ip_blacklist() -> [inet:ipv4_address()]).
+bind_ip_blacklist() ->
+    IpStrs = application:get_env(?APP, bind_ip_blacklist, []),
+    lists:map(fun spartan_app:parse_ipv4_address/1, IpStrs).
+
+-spec(bind_ips() -> [inet:ipv4_address()]).
 bind_ips() ->
     case application:get_env(?APP, bind_ips, []) of
         [] ->
             IPs0 = bind_ips2(),
-            IPs1 = lists:usort(IPs0),
-            application:set_env(?APP, bind_ips, IPs1),
-            IPs1;
+            lager:info("found ips: ~p", [IPs0]),
+            IPs1 = bind_ip_blacklist(),
+            lager:info("blacklist ips: ~p", [IPs1]),
+            IPs2 = lists:subtract(IPs0, IPs1),
```

Comment:
maybe ordsets:subtract? lists:subtract could be buggy because it removes only one occurrence